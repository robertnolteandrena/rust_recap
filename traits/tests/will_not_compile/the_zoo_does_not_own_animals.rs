use spectral::assert_that;
use traits_lib::{Goat, HasName, Zoo};

fn does_the_zoo_own_animals_quesionmark() {
    let mut zoo = Zoo { animals: vec![] };
    {
        let goat = Goat;
        zoo.animals.push(&goat);
    }

    let zoo_name = zoo.name();
    assert_that!(zoo_name.as_ref()).is_equal_to("Zoo of: Goat");
}
fn main() {}
