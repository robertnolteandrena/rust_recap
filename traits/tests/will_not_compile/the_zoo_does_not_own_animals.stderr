error[E0597]: `goat` does not live long enough
  --> tests/will_not_compile/the_zoo_does_not_own_animals.rs:8:26
   |
7  |         let goat = Goat;
   |             ---- binding `goat` declared here
8  |         zoo.animals.push(&goat);
   |                          ^^^^^ borrowed value does not live long enough
9  |     }
   |     - `goat` dropped here while still borrowed
10 |
11 |     let zoo_name = zoo.name();
   |                    --- borrow later used here
